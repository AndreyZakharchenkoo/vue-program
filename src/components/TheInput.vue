<template>
  <input
    type="text"
    class="input"
  />
</template>

<script>
export default {
  name: 'TheInput'
}
</script>

<style lang="scss" scoped>
.input {
  display: inline-flex;
  width: 100%;
  height: 55px;
  padding: 0 20px;
  border-radius: 4px;
  border: 2px solid transparent;
  background-color: rgba(29, 29, 29, 0.8);
  color: #fff;
  font-family: inherit;
  font-size: 20px;
  line-height: 1.5;
  font-weight: 500;
  transition: .2s;

  &:focus {
    outline: none;
    border-color: #f65261;
  }
}
</style>
