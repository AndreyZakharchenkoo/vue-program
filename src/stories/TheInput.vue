<template>
  <input
    type="text"
    class="input"
    v-bind="$attrs"
    :value="value"
    v-on="inputListeners"
  />
</template>

<script>
export default {
  name: 'TheInput',
  props: ['value'],
  inheritAttrs: true,
  computed: {
    inputListeners: function () {
      const vm = this

      return Object.assign({},
        this.$listeners,
        {
          input: function (event) {
            vm.$emit('input', event.target.value)
          }
        }
      )
    }
  }
}
</script>

<style scoped>
.input {
  display: inline-flex;
  width: 100%;
  height: 55px;
  padding: 0 20px;
  border-radius: 4px;
  border: 2px solid transparent;
  background-color: rgba(29, 29, 29, 0.8);
  color: #fff;
  font-family: inherit;
  font-size: 20px;
  line-height: 1.5;
  font-weight: 500;
  transition: .2s;
}

.input:focus {
  outline: none;
  border-color: #f65261;
}
</style>
