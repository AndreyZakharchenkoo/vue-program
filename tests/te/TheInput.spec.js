import { shallowMount } from '@vue/test-utils';
import TheInput from '@/stories/TheInput.vue';

describe('TheInput.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(TheInput, {});
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('Renders correctly', () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
