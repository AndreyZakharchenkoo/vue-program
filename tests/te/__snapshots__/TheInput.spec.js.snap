// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheInput.vue Renders correctly 1`] = `
<input
  class="input"
  type="text"
/>
`;
